src/cloud/provision/freedom-module.ts

```diff
@@ -0,0 +1,362 @@
+/// <reference path='../../../../third_party/typings/node/node.d.ts' />
+/// <reference path='../../../../third_party/typings/es6-promise/es6-promise.d.ts' />
+/// <reference path='../../../../third_party/typings/freedom/freedom-module-env.d.ts' />
+
+const forge = require("forge-min");
+
+const POLL_TIMEOUT: number = 5000; //milliseconds
+
+const STATUS_CODES: { [k: string]: string; } = {
+  "START": "Starting provisioner",
+  "OAUTH_INIT": "Initializing oauth flow",
+  "OAUTH_ERROR": "Error getting oauth token",
+  "OAUTH_COMPLETE": "Got oauth token",
+  "SSHKEY_RETRIEVED": "Retrieved SSH keys from storage",
+  "SSHKEY_GENERATED": "Generated new SSH keys",
+  "CLOUD_FAILED": "Failed to complete cloud operation",
+  "CLOUD_INIT_ADDKEY": "Starting to add SSH key to cloud account",
+  "CLOUD_DONE_ADDKEY": "Done adding SSH key to cloud account",
+  "CLOUD_INIT_VM": "Starting to provision VM",
+  "CLOUD_DONE_VM": "Done provisioning VMs",
+  "CLOUD_WAITING_VM": "Waiting on VM",
+};
+
+const ERR_CODES: { [k: string]: string; } = {
+  "VM_DNE": "VM does not exist",
+  "CLOUD_ERR": "Error from cloud provider"
+};
+
+const REDIRECT_URIS: [string] = [
+  "https://pjpcdnccaekokkkeheolmpkfifcbibnj.chromiumapp.org"
+  //  "http://localhost:10101"
+];
+
+interface KeyPair {
+  private: string;
+  public: string;
+}
+
+class Provisioner {
+  private dispatch: Function;
+  private state: any;
+
+  constructor(dispatchEvent: Function, state: any) {
```

Comment:
We find the automatic constructor assignment stuff useful, i.e.:
```typescript
constructor(private dispatchEvent_: Function, private state_: any) {}
```

Saves quite a few lines.